src/partisan/irods.py

```diff
@@ -2837,6 +2838,11 @@ def put(
             local_path: The local path of a directory to put into iRODS at the path
                 specified by this collection.
             recurse: Recurse through subdirectories.
+            filter_fn: A predicate accepting a single pathlib.Path argument to which
+                each local path (directories and files) will be passed before putting
+                into iRODS. If the predicate returns True, the path will be filtered
+                i.e. not be put into iRODS. Filtering collections will result in them
+                being pruned. Filtering data objects will result in them being skipped.
```

Comment:
I found this wording confusing, I think one issue is that it implies remote action, when it is actually the directories being pruned from the walk (local action).  Also drawing a specific distinction between pruning a directory and skipping a file initially caused me issues, but I see why you did it.

Reply:
Do you have a suggestion to clarify the wording?

Reply:
Perhaps something like "Filtering on an iRODS path will remove the associated local path from the discovery process"? I think there's a better word than associated, but I can't think of it at the moment...

Reply:
I've changed it to describe it in terms of directories and files, as that's what's actually being filtered.